package rec.util.file;

message VirtualFile {
  enum Type {
    NONE = 0;
    CD = 1;
    MUSIC = 2;
    USER = 3;
    VOLUME = 4;
  };

  enum Status {
    ONLINE = 1;
    OFFLINE = 2;
    DISK_OPEN = 3;
    WRITEABLE_DISK = 4;
    NO_DISK = 5;
    UNKNOWN = 6;
  };

  optional Type type = 1 [default = NONE];
  optional string volume_name = 2;
  repeated string path = 3;
  optional Status status = 4 [default = ONLINE];
};

message VirtualFileList {
  repeated VirtualFile file = 1;
};